src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ReadyToRunHelperNode.cs

```diff
@@ -133,5 +156,75 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
                 return null;
             }
         }
+
+        protected override void OnMarked(NodeFactory factory)
+        {
+            switch (_id)
+            {
+                case ReadyToRunHelperId.GenericLookupFromThis:
+                case ReadyToRunHelperId.GenericLookupFromDictionary:
+                    {
+                        // When the helper call gets marked, ensure the generic layout for the associated dictionaries
+                        // includes the signature.
+                        var lookupInfo = (GenericLookupDescriptor)_target;
+                        factory.GenericDictionaryLayout(lookupInfo.CanonicalOwner).EnsureEntry(lookupInfo.Signature);
```

Comment:
Note: I could opt to do this the same way as we do VTable slots (with a separate GenericDictionarySlotUse node that is analogous to the VirtualMethodUse node), but I'm not sure it would buy us anything. Thoughts?